spec/graph_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'spec_helper'
+require 'ronin/recon/graph'
+
+describe Ronin::Recon::Graph do
+  subject { described_class.new }
+  
+  describe "#initialize" do
+    it "must initialize #nodes and #edges" do
+      expect(subject.nodes).to eq(Set.new)
+      expect(subject.edges).to eq({})
+    end
+  end
+  
+  describe "#add_node" do
+    context "when value node was successfully added" do
+      it "must return true" do
+        expect(subject.add_node("node_value")).to be(true)
```

Comment:
For realness, could you use a `Values::IP` or `Values::Host` object instead of a primitive value? Probably could define `let(:value1) { ... }` and `let(:value2) { ... }` and reuse them in the specs.